src/snowflake/connector/cursor.py

```diff
@@ -1175,15 +1175,18 @@ def scroll(self, value, mode="relative"):
             },
         )
 
-    def reset(self):
+    def reset(self, closing=False):
```

Comment:
I also fins this name misleading. If I want to not close the cursor why would I want the `rowcount` to be reset?
Also if we want `rowcount` to stay after a `reset` call do we want `_result_set` to stick around too?

Reply:
Thank you for reviewing this Mark,

> I also find this name misleading.

To clarify, is this note regarding the use of the name `closing` for the optional argument?

> If I want to not close the cursor why would I want the rowcount to be reset?

The existing behavior of `cursor.reset()` is mainly around the correctness of multiple executions performed over a single cursor. The reset call ensures that upon a failure of the new execution the past attributes are not returned back.

In the case of `cursor.close()` calling `cursor.reset()` internally, no further executions will be permitted so it appears OK to retain the current `rowcount`.

> Also if we want `rowcount` to stay after a `reset` call do we want `_result_set` to stick around too?

No, the spec does strictly say no more operations should be permitted after a `cursor.close()` and I do not see any part in SQLAlchemy accessing data or requesting executions after closes. It appears to only be the cursor's attributes (`rowcount` and `description`) that appear accessed after `cursor.close()`.